src/main/java/cz/sparko/boxitory/service/filesystem/FilesystemHashStore.java

```diff
@@ -0,0 +1,131 @@
+package cz.sparko.boxitory.service.filesystem;
+
+import cz.sparko.boxitory.service.HashService.HashAlgorithm;
+import cz.sparko.boxitory.service.HashStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Stores/reads calculated checksums on/from filesystem. Hash file is named
+ * {@code {box_filename}.{hash_algorithm_extension}} and has one-line content {@code {calculated_hash}  {box_filename}}.
+ * Checksum files can be created manually, but must have proper format.
+ */
+public class FilesystemHashStore implements HashStore {
+    private static final Logger LOG = LoggerFactory.getLogger(FilesystemHashStore.class);
+
+    /**
+     * Store checksum on filesystem, beside box file with appended extension derived from
+     * {@link HashAlgorithm}. When already exists, original file is not replaced.
+     * When file can't be created for any reason (e.g. write permissions), error is logged but no exception raised.
+     *
+     * @param box       absolute filesystem path to the box
+     * @param hash      calculated hash
+     * @param algorithm algorithm of hash
+     * @throws IllegalStateException when file provided by {@code box} does not exists
+     */
+    @Override
+    public void persist(String box, String hash, HashAlgorithm algorithm) {
+        File boxFile = checkBoxFileExists(box);
+
+        if (algorithm == HashAlgorithm.DISABLED) {
+            LOG.debug("Hash algorithm [{}]. Nothing to persist.", algorithm);
+            return;
```

Comment:
IMHO I think that it is better to throw exception instead of returning nothing. But it is probably the way of personal approach. Or maybe to change return typo to bool and return in failed cases false.

Reply:
you're right. Trying to persist checksum with disabled checksums is invalid state. fixed